docs/docs/examples/repl.md

```diff
@@ -34,43 +34,57 @@ Steps
 
         opa run
 
-    Without any data, you can experiment with simple boolean expressions to get the hang of it:
-
-        > a = 1, b = 2, a != b
-        +---+---+
-        | A | B |
-        +---+---+
-        | 1 | 2 |
-        +---+---+
-        > a = 1, b = 2, a = b
+    Without any data, you can experiment with simple expressions to get the hang of it:
+
+        > true
+        true
+        > 3.14
+        3.14
+        > ["hello", "world"]
+        [
+          "hello",
+          "world"
+        ]
+
+    You can also test simple boolean expressions:
+
+        > true = false
         false
-        > 10 > 9
+        > 3.14 > 3
         true
-        > a = [1,2,3,4], a[i] > 2
-        +-----------+---+
-        |     A     | I |
-        +-----------+---+
-        | [1,2,3,4] | 2 |
-        | [1,2,3,4] | 3 |
-        +-----------+---+
+        > "hello" != "goodbye"
+
+    When you enter expressions into the REPL, you are effectively running *queries* against OPA. The REPL output shows the values of variables in the expression that make the query **true**. If there is no set of variables that would make the query true, the REPL prints **false**. If there are no variables in the query and the query evaluates successfully, then the REPL just prints **true**.
+
+    Most REPLs let you define variables that you can reference later on. OPA allows you to do something similiar. For example, we can define a "pi" constant as follows:
+
+        > pi = 3.14
 
-    When you enter expressions into the REPL, you are effectively running *queries* against OPA. The REPL output shows the values of variables in the expression that make the query **true**. If there is not set of variables that would make the query true, the REPL prints **false**. If there are no variables in the query and the query evaluates successfully, then the REPL just prints **true**.
+    Once "pi" is defined, you query for the value and write expressions in terms of it:
 
-1. In addition to running queries, the REPL also lets you define rules:
+        > pi
+        3.14
+        > pi = 3
+        false
+        > pi = 3.14
+        true
 
```

Comment:
This is surprising: the pi = 3.14 is an assignment but pi = 3 is a test (b/c pi is already assigned).  What we're saying is that there's no way to reassign the value for pi.  That makes sense if we think about the repl as incrementally creating a module.  But I would imagine most people would expect pi = 3 to reassign p to 3, not test if it equals 3 (which for us would mean replacing the original rule in the module with a new one).  

For the case of p=3 and then later p=5 I'd imagine the expected behavior is to reassign p to 5.  But  if we assign p = 3 and q = 4 and then write p = q it's totally unclear if the intent is that it is a check for equality or an assignment.  

For partial definitions, I think it's reasonable for p[x] :- q[x] and then p[x] :- r[x] to be additive.  One thing we might want is to have an 'undefine p' command for just the repl that erases all the rules for p.  Perhaps we could use the same command to help with the p=3.14 p=3 issue above, so that at least there's a way of reassigning p.

If we did that, I'd think we'd want a sentence or two explaining the issue....

Most REPLs let you define variables you can reference later on.  For example, we can define a constant "pi" as follows:

> pi = 3.14

Once "pi" is defined, you query for the value and write expressions using it:

> pi
3.14
> pi > 3
true

One thing to watch out for in the REPL is that = is used both for assigning variables values and for testing the value of variables.  For example p = q sometimes assigns p the value of q and sometimes checks if the values of p and q are the same.  The REPL decides between assignment and test based on whether p already has a value or not.  If p has a value, p = q is a test (returning true or false), and if p has no value p = q is an assignment.  To unset a value for a variable, use the 'unset' command.  (This ambiguity is only really an issue in the REPL--when writing policy the duality of = is actually beneficial.)

> unset p